src/lib.rs

```diff
@@ -341,4 +333,10 @@ pub trait AsyncConnection: SimpleAsyncConnection + Sized + Send {
     fn transaction_state(
         &mut self,
     ) -> &mut <Self::TransactionManager as TransactionManager<Self>>::TransactionStateData;
+
+    #[doc(hidden)]
+    fn _silence_lint_on_execute_future(_: Self::ExecuteFuture<'_, '_>) {}
```

Comment:
Can you add a comment to these function explaining why they exist + linking to the relevant rust issues?